src/HamamatsuCamera.cpp

```diff
@@ -2977,4 +2977,290 @@ void Camera::setHighDynamicRangeEnabled(const bool & in_enabled)
+
+        if((err != DCAMERR_INVALIDPROPERTYID)&&(err != DCAMERR_NOTSUPPORT))
+        {
+            THROW_HW_ERROR(Error) << "Unable to retrieve the output trigger kind attribute";
+        }
+    }    
+    else
+    {
+        //Get the ARRAYELEMENT size to ensure that the given channel is reachable
+        err = dcamprop_getvalue(m_camera_handle, basepropattr.iProp_NumberOfElement, &kindArraySize);
+	    if (!failed(err) && channel < kindArraySize)
+	    {
+            //Get the channel kind value
+            double tmp = 99;
+			err = dcamprop_getvalue(m_camera_handle, basepropattr.iProp + channel * basepropattr.iPropStep_Element, &tmp);
+
+            if(!failed(err)){
+                
+                
+                int32 value = static_cast<int32>(tmp);
+
+                switch(value) 
+                {
+                    case DCAMPROP_OUTPUTTRIGGER_KIND__LOW              : kind = Camera::Output_Trigger_Kind_Low             ; break;
+                    case DCAMPROP_OUTPUTTRIGGER_KIND__EXPOSURE         : kind = Camera::Output_Trigger_Kind_Global_Exposure ; break;
+                    case DCAMPROP_OUTPUTTRIGGER_KIND__PROGRAMABLE      : kind = Camera::Output_Trigger_Kind_Programmable    ; break;
+                    case DCAMPROP_OUTPUTTRIGGER_KIND__TRIGGERREADY     : kind = Camera::Output_Trigger_Kind_TriggerReady    ; break;
+                    case DCAMPROP_OUTPUTTRIGGER_KIND__HIGH             : kind = Camera::Output_Trigger_Kind_High            ; break;
+                    default: break; // result will be Output_Trigger_Kind_Not_Supported
+
+                }
+            } //else TODO
+        }
+    }
+    
+    return kind;
+}
+
+//=============================================================================
+// OUTPUT TRIGGER POLARITY
+//=============================================================================
+//-----------------------------------------------------------------------------
+/// Return the output trigger polarity of given channel by the current detector
+//-----------------------------------------------------------------------------
+enum Camera::Output_Trigger_Polarity Camera::getOutputTriggerPolarity(int channel)
+{
+    
+    DEB_MEMBER_FUNCT();
+
+    DEB_TRACE() << "Camera::getOutputTriggerPolarity(int channel) : channel = " << channel ;
+
+    enum Output_Trigger_Polarity result = Output_Trigger_Polarity::Output_Trigger_Polarity_Not_Supported;
+
+    DCAMERR err;
+    double polarityArraySize = 0;
+    enum Output_Trigger_Polarity polarity = Camera::Output_Trigger_Polarity_Not_Supported;
+
+	// get property attribute that contains the Polarity (that may be an array)
+	DCAMPROP_ATTR	basepropattr;
+	memset(&basepropattr, 0, sizeof(basepropattr));
+	basepropattr.cbSize = sizeof(basepropattr);
+	basepropattr.iProp = DCAM_IDPROP_OUTPUTTRIGGER_POLARITY;
+	err = dcamprop_getattr(m_camera_handle, &basepropattr);

```

Comment:
1/ could this code be replaced by the getPropertyData function ?
2/ fix the formatting: ie start is not aligned